### ISSUE TITLE
Tilde expansion

### ISSUE THREAD

**Issue by mmstick** [MovingtoMars/liner#34]
Something else I've noticed with completions in the Ion shell is that if your attempt to cd to a directory starting with the `~` character. Is there a way tto either supply the expanded form to liner or have liner expand it?

**Comment by MovingtoMars** [comment]
Yeah, this is a good idea. I imagine we'll need a `home_directory` member for FilenameCompleter.

Or, perhaps a callback so we can support more advanced file expansions more easily (other users' home directories etc)

**Comment by mmstick** [comment]
Ion has this working perfectly without any changes to Liner, but I was unable to do this without the usage of `unsafe`, so we might want to have a better solution in the long run.
